tools/fast/tests/retrieval_test.go

```diff
@@ -108,18 +108,87 @@ func TestRetrieval(t *testing.T) {
 
 	// Everyone needs FIL to deal with gas costs and make sure their wallets
 	// exists (sending FIL to a wallet addr creates it)
-	err = series.SendFilecoinDefaults(ctx, genesis, miner, 100000)
+	err = series.SendFilecoinDefaults(ctx, genesis, miner, 1000)
 	require.NoError(t, err)
 
-	err = series.SendFilecoinDefaults(ctx, genesis, client, 100000)
+	err = series.SendFilecoinDefaults(ctx, genesis, client, 1000)
 	require.NoError(t, err)
 
-	// Start retrieval
-	// Start retrieval
-	// Start retrieval
+	RunRetrievalTest(ctx, t, miner, client, sectorSize)
+}
+
+// TestRetrieval exercises storing and retreiving with the filecoin protocols
+func TestNetworkRetrieval(t *testing.T) {
+	tf.FunctionalTest(t)
+
+	// Skip the test so it doesn't run
+	t.SkipNow()
+
+	blocktime := time.Second * 30
```

Comment:
It seems to me like we could reduce this. Even though a functional test is known to be "slow", when running locally there is still great advantage to it being faster.

Reply:
This is a network retrieval, so it should be running the same blocktime as the network. I don't know how we could reduce it. Once we have enforcement in consensus it can't be changed anyways.

Reply:
What does "network retrieval" mean, and what's a retrieval that doesn't involve a network?

I might be misunderstanding the intent. The name and comment make it sound like this should be fairly generic, not coupled to any particular network configuration. And if not, I suggest we should refactor to separate the generic from the specific so that the same test code can operate in multiple environments. I have just now realised that this bears a very strong similarity to code immediately above it. Is this intended to be essentially the same test but against a network of already-running nodes? The comments don't really differentiate. Can we factor that out (more)?

Reply:
Ya, I forgot to update those comments. I've updated them now.

This test is refactored to be generic as it can be at the moment, and may become more generic as FAST matures. The `RunRetrievalTest` test at the bottom of this file is the actual test which takes in a set of resources. These top functions now `TestRetrievalLocalNetwork` and `TestRetrievalDevnet` do the required setup that is not common between the two.

The `TestRetrievalLocalNetwork` is probably the test you want to be looking at. It uses a 5 second block time, which might be able to be reduced but I haven't played with the number much to determine if there is a lower bound. 5 second seems to work consistently though.

Reply:
More of this could be made generic once we figure out the correct way to deal with the `GetFunds` concept. Comment thread above talks about this a bit.